pulp_ansible/tests/functional/api/collection/v2/test_proxy.py

```diff
@@ -0,0 +1,94 @@
+import pytest
+from pulp_smash.pulp3.bindings import monitor_task, PulpTaskError
+
+from pulpcore.client.pulp_ansible import (
+    AnsibleRepositorySyncURL,
+)
+
+
+def _sync_and_assert(
+    remote, ansible_repo, ansible_repo_api_client, ansible_collection_version_api_client
+):
+    body = AnsibleRepositorySyncURL(remote=remote.pulp_href)
+    monitor_task(ansible_repo_api_client.sync(ansible_repo.pulp_href, body).task)
+
+    # Check content is present, but no artifacts are there
+    content_response = ansible_collection_version_api_client.list()
+    assert content_response.count == 1
+
+
+@pytest.mark.parallel
+def test_sync_through_http_proxy(
+    ansible_repo,
+    gen_ansible_collection_remote,
+    ansible_repo_api_client,
+    ansible_collection_version_api_client,
+    http_proxy,
+):
+    """
+    Test syncing through a http proxy.
+    """
+    ansible_remote = gen_ansible_collection_remote(
+        url="https://galaxy.ansible.com",
+        requirements_file="collections:\n  - testing.k8s_demo_collection",
+        sync_dependencies=False,
+        proxy_url=http_proxy.proxy_url,
+    )
+
+    _sync_and_assert(
+        ansible_remote, ansible_repo, ansible_repo_api_client, ansible_collection_version_api_client
+    )
```

Comment:
Just wondering, whether this could be handled with `mark.parameterize`. And it's probably a no.

Reply:
I'm not understanding the idea.

Reply:
Probably could be done with parameterize. https://docs.pytest.org/en/6.2.x/parametrize.html#pytest-mark-parametrize. Not sure how clean it would be.

Reply:
I'm familiar with parameterize, but I'm not sure what the we're parameterize with this idea.

Reply:
The three tests are identical besides the arguments to the remote. Those arguments could be parameterize, even including the case that fails with: https://docs.pytest.org/en/6.2.x/reference.html#pytest.param, resulting with one test that gets ran 3 times. Not sure how cleanly you could make it though...

Reply:
Mmmm I see. Yeah I tried to do something like this actually, but I ended up backing it out. It's a bit tricky because in addition to one of the tests expecting a failure and the other two not, the fixture type used is subtly different too. Two use `http_proxy_with_auth` and the other uses `http_proxy`. The pytest way to paramaterize fixtures is with the `indirect=True` feature, but to do that you have to create an additional fixture that operates kind of like a switcher. I ended up going with this because the other option seemed more complicated. I very much want to use the parameterize though, I haven't really gotten to yet.